src/test/regress/sql/citus_local_tables.sql

```diff
@@ -0,0 +1,368 @@
+
+  -- print sequence ownerships
+  -- show that the only internal sequence is on col1 and it is owned by shell relation
+  SELECT s.relname as sequence_name, t.relname, a.attname
+  FROM pg_class s
+    JOIN pg_depend d on d.objid=s.oid and d.classid='pg_class'::regclass and d.refclassid='pg_class'::regclass
+    JOIN pg_class t on t.oid=d.refobjid
+    JOIN pg_attribute a on a.attrelid=t.oid and a.attnum=d.refobjsubid
+  WHERE s.relkind='S' and s.relname like 'citus_local_table_3%' ORDER BY 1,2;
+ROLLBACK;
+
+-- test foreign tables using fake FDW --
+
+CREATE FOREIGN TABLE foreign_table (
+  id bigint not null,
+  full_name text not null default ''
+) SERVER fake_fdw_server OPTIONS (encoding 'utf-8', compression 'true');
+
+-- observe that we do not create fdw server for shell table, both shard relation
+-- & shell relation points to the same same server object
+SELECT create_citus_local_table('foreign_table');
+
+-- drop them for next tests
+DROP TABLE citus_local_table_1, citus_local_table_2, distributed_table;
+
+-- create test tables
+CREATE TABLE citus_local_table_1 (a int primary key);
+SELECT create_citus_local_table('citus_local_table_1');
+
+CREATE TABLE citus_local_table_2 (a int primary key);
+SELECT create_citus_local_table('citus_local_table_2');
+
+CREATE TABLE local_table (a int primary key);
+
+CREATE TABLE distributed_table (a int primary key);
+SELECT create_distributed_table('distributed_table', 'a');
+
+CREATE TABLE reference_table (a int primary key);
+SELECT create_reference_table('reference_table');
+
+-- show that colociation of citus local tables are not supported for now
+-- between citus local tables
+SELECT mark_tables_colocated('citus_local_table_1', ARRAY['citus_local_table_2']);
+
+-- between citus local tables and reference tables
+SELECT mark_tables_colocated('citus_local_table_1', ARRAY['reference_table']);
+SELECT mark_tables_colocated('reference_table', ARRAY['citus_local_table_1']);
+
+-- between citus local tables and distributed tables
+SELECT mark_tables_colocated('citus_local_table_1', ARRAY['distributed_table']);
+SELECT mark_tables_colocated('distributed_table', ARRAY['citus_local_table_1']);
+
+-- upgrade_to_reference_table is not supported
+SELECT upgrade_to_reference_table('citus_local_table_1');
+-- master_create_empty_shard is not supported
+SELECT master_create_empty_shard('citus_local_table_1');
+-- get_shard_id_for_distribution_column is supported
+SELECT get_shard_id_for_distribution_column('citus_local_table_1', 'not_checking_this_arg_for_non_dist_tables');
+SELECT get_shard_id_for_distribution_column('citus_local_table_1');
+-- master_copy_shard_placement is not supported
+SELECT master_copy_shard_placement(shardid, 'localhost', :master_port, 'localhost', :worker_1_port, true)
+FROM (SELECT shardid FROM pg_dist_shard WHERE logicalrelid='citus_local_table_1'::regclass) as shardid;
+-- undistribute_table is supported
+BEGIN;
+  SELECT undistribute_table('citus_local_table_1');
```

Comment:
We should also make sure that `master_copy_shard_placement()` has an explicit error check for  citus local tables. The correct place could be  `ErrorIfTableCannotBeReplicated()`


If you create a citus local table with statement replication, it gives a weird error now

```
SELECT master_copy_shard_placement(102082, 'localhost', 5432, 'localhost', 9700);
ERROR:  could not find placement matching "localhost:9700"
HINT:  Confirm the placement still exists and try again.
```

Reply:
this should be outdated, I added necessary changes for that in
https://github.com/citusdata/citus/blob/e36b26ea8bdc946ca934f89ecb47bcda4f75d2da/src/backend/distributed/operations/repair_shards.c#L186-L213